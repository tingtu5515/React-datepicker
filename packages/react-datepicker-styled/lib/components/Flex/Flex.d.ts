import {
  AlignItemsProps,
  FlexDirectionProps,
  FlexWrapProps,
  JustifyContentProps,
  SpaceProps,
  GridAreaProps,
} from 'styled-system'
interface FlexProps
  extends SpaceProps,
    FlexWrapProps,
    FlexDirectionProps,
    AlignItemsProps,
    GridAreaProps,
    JustifyContentProps {}
declare const Flex: import('styled-components').StyledComponent<'div', any, FlexProps, never>
export default Flex
