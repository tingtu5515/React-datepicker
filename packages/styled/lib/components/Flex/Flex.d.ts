import { AlignItemsProps, FlexDirectionProps, FlexWrapProps, JustifyContentProps, SpaceProps, GridAreaProps, HeightProps, WidthProps } from 'styled-system';
interface FlexProps extends SpaceProps, FlexWrapProps, FlexDirectionProps, AlignItemsProps, GridAreaProps, HeightProps, WidthProps, JustifyContentProps {
}
declare const Flex: import("styled-components").StyledComponent<"div", any, FlexProps, never>;
export default Flex;
