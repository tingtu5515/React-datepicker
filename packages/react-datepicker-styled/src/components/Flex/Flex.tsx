import {
  alignItems,
  AlignItemsProps,
  flex,
  flexDirection,
  FlexDirectionProps,
  flexWrap,
  FlexWrapProps,
  justifyContent,
  JustifyContentProps,
  space,
  SpaceProps,
  gridArea,
  GridAreaProps,
} from 'styled-system'
import styled from 'styled-components'

interface FlexProps
  extends SpaceProps,
    FlexWrapProps,
    FlexDirectionProps,
    AlignItemsProps,
    GridAreaProps,
    JustifyContentProps {}
const Flex = styled('div')<FlexProps>`
  display: flex;

  ${space}
  ${flex}
  ${flexWrap}
  ${flexDirection}
  ${alignItems}
  ${justifyContent}
  ${gridArea}
`

export default Flex
