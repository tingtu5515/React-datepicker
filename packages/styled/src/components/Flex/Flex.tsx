import {
  alignItems,
  AlignItemsProps,
  flex,
  FlexProps,
  flexDirection,
  FlexDirectionProps,
  flexWrap,
  FlexWrapProps,
  justifyContent,
  JustifyContentProps,
  space,
  SpaceProps,
  gridArea,
  GridAreaProps,
  height,
  HeightProps,
  width,
  WidthProps,
  compose,
} from 'styled-system'
import styled from 'styled-components'

interface FlexComponentProps
  extends SpaceProps,
    FlexWrapProps,
    FlexDirectionProps,
    AlignItemsProps,
    GridAreaProps,
    HeightProps,
    WidthProps,
    FlexProps,
    JustifyContentProps {}

const composeFlexStyles = compose(
  space,
  flex,
  flexWrap,
  flexDirection,
  alignItems,
  justifyContent,
  gridArea,
  height,
  width,
)

const Flex = styled('div')<FlexComponentProps>`
  display: flex;
  ${composeFlexStyles}
`

export default Flex
