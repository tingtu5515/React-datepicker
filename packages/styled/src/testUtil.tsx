import React from 'react'
import {render} from '@testing-library/react'
import DatepickerContext, {datepickerContextDefaultValue} from './context/datepickerContext'

function renderWithProviders(ui: any, options?: any, contextValues?: Record<string, unknown>) {
  return render(
    <DatepickerContext.Provider value={{...datepickerContextDefaultValue, ...contextValues}}>
      {ui}
    </DatepickerContext.Provider>,
    options,
  )
}

export * from '@testing-library/react'
export {renderWithProviders as render}
