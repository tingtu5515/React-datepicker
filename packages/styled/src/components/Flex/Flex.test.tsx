import React from 'react'
import {render} from '../../testUtil'
import Flex from '.'

test('should match snapshot', () => {
  const {container} = render(
    <Flex
      height="50px"
      width="50px"
      gridArea="test"
      justifyContent="center"
      alignItems="center"
      flexDirection="column"
      flexWrap="wrap"
      m="30px"
    >
      <div>test1</div>
      <div>test2</div>
      <div>test3</div>
      <div>test4</div>
      <div>test5</div>
    </Flex>,
  )
  expect(container).toMatchSnapshot()
})
