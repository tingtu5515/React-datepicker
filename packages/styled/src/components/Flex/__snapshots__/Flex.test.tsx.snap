// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 50px;
  width: 50px;
  grid-area: test;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin: 30px;
}

<div>
  <div
    class="c0"
    height="50px"
    width="50px"
  >
    <div>
      test1
    </div>
    <div>
      test2
    </div>
    <div>
      test3
    </div>
    <div>
      test4
    </div>
    <div>
      test5
    </div>
  </div>
</div>
`;
